harness/determined/pytorch/deepspeed/dsat/_dsat_search_method.py

```diff
@@ -351,7 +351,7 @@ def should_shutdown(self) -> bool:
         if self.early_stopping_triggered:
             logging.info("Early stopping criteria met, searcher will shut down.")
             return True
-        elif self.all_trials_created and self.all_trials_closed_or_errored:
+        elif self.all_trials_closed_or_errored:
```

Comment:
Was having both checks leading to bugs?  I thought we'd want both conditions.

I don't *think* this should happen, but my specific concern was that there might be scenarios in which only a subset of all the trials we want to run have been created and they all close or error out before any follow on ones can be created, which would set `all_trials_closed_or_errored =True` while `all_trials_created` is still `False`

Reply:
Yes, and it's because of the case you are talking about, it has not yet made all the trials it thought it would, but all trials have failed and it should stop. This happens in the case that the model info profile run is failing and it is fixed primarily by making sure that in all cases of failure, we close the experiment, which will trigger this check for whether all the trials are closed or failed

Reply:
I modified the tests so that we are closing when exiting early which to the best of my understanding is what we are doing on the master side as well. I reverted this particular change